pkg/distributor/distributor.go

```diff
@@ -1036,6 +1051,10 @@ func (d *Distributor) prePushHaDedupeMiddleware(next PushFunc) PushFunc {
 		} else {
 			// If there wasn't an error but removeReplica is false that means we didn't find both HA labels.
 			d.nonHASamples.WithLabelValues(userID).Add(float64(numSamples))
+			if d.limits.TrackHAFailures(userID) {
+				d.failedHARequests.WithLabelValues(userID, metricName, keepReplicaReason).Inc()
```

Comment:
I think you need to copy `metricName`, for this reason:
```
// Strings in pushReq may be pointers into the gRPC buffer which will be reused, so must be copied if retained.
```

Reply:
Good catch, thanks!